datastream-server/src/main/java/com/linkedin/datastream/server/EventProducer.java

```diff
@@ -200,13 +200,24 @@ private void validateEventRecord(DatastreamProducerRecord record) {
     }
   }
 
+  @Override
+  public void broadcast(DatastreamProducerRecord record, SendCallback sendBroadcastCallback) {
+    helperSendOrBroadcast(record, sendBroadcastCallback, null, true);
+  }
+
+  @Override
+  public void send(DatastreamProducerRecord record, SendCallback sendCallback) {
+    helperSendOrBroadcast(record, null, sendCallback, false);
+  }
+
   /**
    * Send the event onto the underlying transport.
    * @param record the datastream event
-   * @param sendCallback the callback to be invoked after the event is sent to the destination
+   * @param sendBroadcastCallback the callback to be invoked after the event is broadcasted to the destination
+   * @param sendEventCallback the callback to be invoked after the event is sent to the destination
    */
-  @Override
-  public void send(DatastreamProducerRecord record, SendCallback sendCallback) {
+  public void helperSendOrBroadcast(DatastreamProducerRecord record, SendCallback sendBroadcastCallback,
```

Comment:
public?

Reply:
Will make it private, thanks for pointing out. This should not be public.